keras_rs/src/utils/ranking_metrics_utils.py

```diff
@@ -0,0 +1,240 @@
+from typing import Callable, Optional
```

Comment:
I'm not a huge fan of the concept of a `utils` folder in general, and I typically prefer the utils files to be in the same folder as where it's used. So for instance, I think this could be in `keras_rs/src/metrics`.

What do you think? Note that keras itself is not consistent on this.